app/controllers/actions/CacheDeleteAction.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import connectors.Cache
+import javax.inject.{Inject, Singleton}
+import models.requests.AuthenticatedRequest
+import play.api.mvc.{ActionFilter, Result}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CacheDeleteActionImpl @Inject()(val dataCacheConnector: Cache, val exc: ExecutionContext) extends CacheDeleteAction {
```

Comment:
In general execution context can be here as implicit. In that case you don't need to pass it manually during invoking remove method

Reply:
Was worried about interfering with the required `override protected def executionContext: ExecutionContext` method.   But your suggestion is a good one so I've done it.